### ISSUE TITLE
macOS snapserver crash on simple "bye" from control client

### ISSUE THREAD

**Issue by airdrummingfool** [badaix/snapcast#238]
I've recently noticed a crash with snapserver, and it's very easy to reproduce on my macOS 10.12.4 machine:

1. start snapserver, no arguments necessary: `./snapserver`
1. connect to the control port: `telnet localhost 1705`
1. Send `bye`

snapserver crashes with the following message:
```
Assertion failed: (ec == 0), function unlock, file /BuildRoot/Library/Caches/com.apple.xbs/Sources/libcxx/libcxx-307.5/src/mutex.cpp, line 48.
Abort trap: 6
```

I just found this, so I'm wondering if it has to do with a recent update to macOS. I tried building various versions from the current tip of `develop` back to `v0.11.0`, and every one I tried crashed with the same message. I also tried building/running on a 10.12.4 machine that had a slightly older version of the Command Line Tools for Xcode, but it had the same issue.

**Comment by badaix** [comment]
I did some quick analysis: the JSON RPC endpoint of your telnet connection deletes itself with the reception of the bye. This somehow leads to the unlock of an already destroyed mutex.
Seems that a slight redesign is needed here.

**Comment by badaix** [comment]
should be fixed b56d656c322f0a26466a28684fffdc9bedb388f0

**Comment by airdrummingfool** [comment]
That fixes it for me, thank you!
